apps/sbbtimetable/sbb_timetable.star

```diff
@@ -0,0 +1,203 @@
+load("render.star", "render")
+load("http.star", "http")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("cache.star", "cache")
+load("time.star", "time")
+load("schema.star", "schema")
+
+# Define some constants
+SBB_URL = "https://fahrplan.search.ch/api/stationboard.json?show_delays=1&transportation_types=train&stop="
+FONT_TO_USE = "CG-pixel-3x5-mono"
+NO_FRAMES_TOGGLE = 20
+COLOR_CATEGORY = {
+    "IC": "#700",
+    "EC": "#700",
+    "IR": "#700",
+    "RE": "#700",
+    "RJX": "#700",
+    "TGV": "#700",
+    "NJ": "#700",
+    "S": "#007",
+    "SN": "#000",
+}
+COLOR_DELAY = "#F00"
+
+def main(config):
+    # Get the config values
+    station = config.get("station", "Bern")
+    skiptime = config.get("skiptime", 0)
+
+    # Check if we need to convert the skiptime
+    if type(skiptime) == "string":
+        skiptime = int(skiptime)
+
+    # Check that the skip time is valid
+    if skiptime < 0:
+        skiptime = 0
+
+    # Check if we have the requested data in the cache
+    resp_cached = cache.get("sbb_%s" % station)
+    if resp_cached != None:
+        # Get the cached response
+        print("Hit! Displaying cached data.")
+        resp = json.decode(resp_cached)
+    else:
+        # Get a new reponse
+        print("Miss! Calling API.")
+        resp = http.get("%s%s" % (SBB_URL, station))
+        if resp.status_code != 200:
+            fail("Request failed with status %d", resp.status_code)
+        cache.set("sbb_%s" % station, resp.body(), ttl_seconds = 120)
```

Comment:
Question: Any idea how many stations there are? We'll hit this API potentially `(24 hours * 60 minutes) / 2 minutes = 720` times the number of stations if this app is popular. At the `10080` limit we can only support 14 stations before we'd need an API key. Seems fine for now, but we'll want to keep an eye on it.

Reply:
I think at the moment there are somewhat over 1000 stations (only train). If, later also bus and ship is added, this number will go up to ~29'000 stations.
It probably depends how many tidbyt users you have in Switzerland :-). But if you hit the rate limit, it should also be possible to increase the ttl value to five minutes.